include/boost/math/interpolators/cubic_b_spline.hpp

```diff
@@ -0,0 +1,292 @@
+// Copyright Nick Thompson, 2017
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This implements the compactly supported cubic b spline algorithm described in
+// "Numerical Analsis, Graduate Texts in Mathematics 181", by Rainer Kress, section 8.3.
+// Splines of compact support are faster to evaluate and are better conditioned than classical cubic splines.
+
+// Let f be the function we are trying to interpolate, and s be the interpolating spline.
+// The routine constructs the interpolant in O(N) time, and evaluating s at a point takes constant time.
+// The order of accuracy depends on the regularity of the f, however, assuming f is
+// four-times continuously differentiable, the error is of O(h^4).
+// In addition, we can differentiate the spline and obtain a good interpolant for f'.
+// The main restriction of this method is that the samples of f must be evenly spaced.
+// Interpolators for non-evenly sampled data are forthcoming.
+// Properties:
+// - s(x_j) = f(x_j)
+// - All cubic polynomials interpolated exactly
+
+#ifndef BOOST_MATH_INTERPOLATORS_CUBIC_B_SPLINE_HPP
+#define BOOST_MATH_INTERPOLATORS_CUBIC_B_SPLINE_HPP
+
+#include <limits>
+#include <cmath>
+#include <vector>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/special_functions/fpclassify.hpp>
+
+namespace boost{ namespace math{
+
+using boost::math::constants::third;
+using boost::math::constants::half;
+
+template<class Real>
+Real b3_spline(Real x)
+{
+    auto sixth = half<Real>()*third<Real>();
```

Comment:
It would be more consistent with other Boost.Math code to use RealType (even if in retrospect Real might be better - RealType is perhaps a tad 'Hungarian' ;-) )